### PR TITLE
doc: a handful of doc improvements

### PR DESCRIPTION
**Pull request by jasnell**
Four separate doc improvements covering various issues. I've bundled these up
as separate commits within a single PR to make it easier to land.

### CODE REVIEW

**Comment by jasnell** [comment]
@joyent/node-tsc ... can I get a quick review on these doc changes?

File: doc/api/buffer.markdown [hunk 1]
```diff
@@ -64,7 +64,7 @@ It can be constructed in a variety of ways.
 
 Allocates a new buffer of `size` octets. Note, `size` must be no more than
 [kMaxLength](smalloc.html#smalloc_smalloc_kmaxlength). Otherwise, a `RangeError`
-will be thrown here.
+will be thrown here. Unlike `ArrayBuffers`, the underlying memory for buffers is not initialized. So the contents of a newly created `Buffer` is unknown. Use `buf.fill(0)`to initialize a buffer to zeroes.
```
Comment (trevnorris) [inline doc/api/buffer.markdown]: Reads well. I'd suggest word wrapping the same as above.

**Comment by trevnorris** [comment]
One nit. Otherwise LGTM.

**Comment by jasnell** [comment]
@trevnorris ... better? (I'll squash down the commits before landing)

**Comment by trevnorris** [comment]
looks great.

**Comment by jasnell** [comment]
Landed in b4a0ca2, 1b027ee, ba6c086, e7f4237, 70efdf3, 500ff56, and b436e59

**Pull request closed without merging**